core/src/main/scripts/importer/validateData.py

```diff
@@ -3229,87 +3263,228 @@ def parse_q_value(self, value, column_number):
+    """Groups multiple treatment response files from a study to ebsure consistency.
+
+    All Validator classes that check validity of different treatment response data
+    types in a study should inherit from this class.
+    """
+    prior_validated_sample_ids = None
+    prior_validated_feature_ids = None
+    prior_validated_header = None
+    REQUIRED_HEADERS = ['treatment_id']
+    OPTIONAL_HEADERS = ['name', 'description', 'url']
+    UNIQUE_COLUMNS = ['treatment_id','name']
+
+    def __init__(self, *args, **kwargs):
+        super(TreatmentWiseFileValidator, self).__init__(*args, **kwargs)
+
+    def parseFeatureColumns(self, nonsample_col_vals):
+        self.checkDifferentNameInDb(nonsample_col_vals)
+        return MultipleDataFileValidator.parseFeatureColumns(self, nonsample_col_vals)
+    
+    def checkDifferentNameInDb(self, nonsample_col_vals):
+        # Check for different combinations of treatment_id and treatment_name
+        # in the database. If true, raise warnings for each discrepancy.
+        nonsample_cols =  self.nonsample_cols
+        if (nonsample_cols.index("name") != None):
+
+            treatmentId = nonsample_col_vals[ nonsample_cols.index("treatment_id") ]
+            fileTreatmentName = nonsample_col_vals[ nonsample_cols.index("name") ]
+            dbTreatmentName = self.portal.treatment_map[treatmentId]["name"]
+
+            if (dbTreatmentName != fileTreatmentName):
+                    self.logger.warning(
+                    "Name " + fileTreatmentName + " for treatment `" + treatmentId
+                    + "` is different from name `"+ dbTreatmentName + "` present in the cBioPortal database. "
+                    + "Treatment names in cBioPortal always reflect treatment names in the last imported study. ",
+                    extra={'line_number': self.line_number,
+                            'cause': fileTreatmentName})
+
+    @classmethod
+    def _get_prior_validated_header(cls):
+        return TreatmentWiseFileValidator.prior_validated_header
+
+    @classmethod
+    def _set_prior_validated_header(cls, header_names):
+        TreatmentWiseFileValidator.prior_validated_header = header_names
+    
+    @classmethod
+    def _get_prior_validated_feature_ids(cls):
+        return TreatmentWiseFileValidator.prior_validated_feature_ids
+
+    @classmethod
+    def _set_prior_validated_feature_ids(cls, feature_ids):
+        TreatmentWiseFileValidator.prior_validated_feature_ids = feature_ids
+
+    @classmethod
+    def _get_prior_validated_sample_ids(cls):
+        return TreatmentWiseFileValidator.prior_validated_sample_ids
+
+    @classmethod
+    def _set_prior_validated_sample_ids(cls, sample_ids):
+        TreatmentWiseFileValidator.prior_validated_sample_ids = sample_ids
+    
+    @classmethod
+    def _get_message_features_do_not_match(cls):
+        return "Treatment feature columns (`treatment_id`, ...) in treatment profile data files are not identical."
```

Comment:
hm...I had to read the code to understand this error..... Maybe one idea would be to add more details to this error message, perhaps in the "cause" section. Something like: "The same set of treatments should be used across the different treatment data files for this study. Check that all treatment ids of one file are also present in all other treatment files".

Reply:
Ok, errors now contain a cause in addition to the original message. For treatments and gene sets these are respectively:

>"message": "Treatment feature columns (`treatment_id`, ...) in treatment profile data files are not identical."
>"cause": "The same set of treatments should be used across the different treatment data files for this study. Please ensure that all treatment id's of one file are present in all other treatment files."

>"message": "Gene sets column in score and p-value file are not equal."
>"cause": "The same set of gene sets should be used in the score and p-value files for this study. Please ensure that all gene set id's of one file are present in the other gene set data file."

Reply:
nice, thanks.